recOrder/tests/mmcore_tests/test_core_func.py

```diff
@@ -0,0 +1,212 @@
+from unittest.mock import MagicMock, Mock, call
+import pytest
+import numpy as np
+from numpy import ndarray
+from typing import Callable, Tuple
+
+# tested components
+from recOrder.io.core_functions import *
+
+
+# TODO: move these to fixture or generate with Hypothesis
+# dynamic range
+TIFF_I_MAX = 2**16
+# image size
+IMAGE_WIDTH = np.random.randint(1, 2**12)
+IMAGE_HEIGHT = np.random.randint(1, 2**12)
+PIXEL_COUNT = IMAGE_HEIGHT * IMAGE_WIDTH
+# serialized image from the pycromanager bridge
+SERIAL_IMAGE = np.random.randint(0, TIFF_I_MAX, size=(PIXEL_COUNT,))
```

Comment:
Fantastic. I think on future iterations we can have a really simple noise model here. No changes needed now.